library/oneview_storage_volume_template_facts.py

```diff
@@ -113,17 +140,23 @@ def __init__(self):
         self.resource_client = self.oneview_client.storage_volume_templates
 
     def execute_module(self):
-
+        ansible_facts = dict(storage_volume_templates=[])
         if self.module.params.get('name'):
             storage_volume_template = self.resource_client.get_by('name', self.module.params['name'])
+            if 'compatibleSystems' in self.options and len(storage_volume_template) > 0:
+                ansible_facts['compatible_systems'] = self.resource_client.get_compatible_systems(
+                    storage_volume_template[0]['uri'])
         else:
             storage_volume_template = self.resource_client.get_all(**self.facts_params)
 
-        ansible_facts = dict(storage_volume_templates=storage_volume_template)
+        ansible_facts['storage_volume_templates'] = storage_volume_template
 
         if 'connectableVolumeTemplates' in self.options:
             ansible_facts['connectable_volume_templates'] = self.resource_client.get_connectable_volume_templates()
 
+        if 'reachableVolumeTemplates' in self.options:
+            ansible_facts['reachable_volume_templates'] = self.resource_client.get_reachable_volume_templates()
```

Comment:
reachable-like methods should accept query params (including the 'networks' one), why are you not passing through the params here?